#! /bin/sh -e

# DP: build libffi without building libgcj

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- configure.in.orig	2004-08-27 10:14:35.000000000 +0200
+++ configure.in	2004-08-28 01:52:00.000000000 +0200
@@ -146,8 +146,7 @@
 host_tools="texinfo byacc flex bison binutils gas ld gcc sid sim gdb make patch prms send-pr gprof etc expect dejagnu ash bash bzip2 m4 autoconf automake libtool diff rcs fileutils shellutils time textutils wdiff find uudecode hello tar gzip indent recode release sed utils guile perl gawk findutils gettext zip fastjar"
 
 # libgcj represents the runtime libraries only used by gcj.
-libgcj="target-libffi \
-	target-boehm-gc \
+libgcj="target-boehm-gc \
 	target-zlib \
 	target-qthreads \
 	target-libjava"
@@ -162,6 +161,7 @@
 		target-libmudflap \
 		target-libgfortran \
 		${libgcj} \
+		target-libffi \
 		target-libobjc \
 		target-libada"
 
--- configure~	2004-08-28 02:38:25.000000000 +0200
+++ configure	2004-08-28 10:57:06.000000000 +0200
@@ -901,8 +901,7 @@
 host_tools="texinfo byacc flex bison binutils gas ld gcc sid sim gdb make patch prms send-pr gprof etc expect dejagnu ash bash bzip2 m4 autoconf automake libtool diff rcs fileutils shellutils time textutils wdiff find uudecode hello tar gzip indent recode release sed utils guile perl gawk findutils gettext zip fastjar"
 
 # libgcj represents the runtime libraries only used by gcj.
-libgcj="target-libffi \
-	target-boehm-gc \
+libgcj="target-boehm-gc \
 	target-zlib \
 	target-qthreads \
 	target-libjava"
@@ -917,6 +916,7 @@
 		target-libmudflap \
 		target-libgfortran \
 		${libgcj} \
+		target-libffi \
 		target-libobjc \
 		target-libada"
 
